stable_audio_tools/models/dit.py

```diff
@@ -187,30 +179,38 @@ def _forward(
             global_embed = timestep_embed
 
         # Add the global_embed to the prepend inputs if there is no global conditioning support in the transformer
-        if prepend_inputs is None:
-            # Prepend inputs are just the global embed, and the mask is all ones
-            prepend_inputs = global_embed.unsqueeze(1)
-            prepend_mask = torch.ones((x.shape[0], 1), device=x.device, dtype=torch.bool)
-        else:
-            # Prepend inputs are the prepend conditioning + the global embed
-            prepend_inputs = torch.cat([prepend_inputs, global_embed.unsqueeze(1)], dim=1)
-            prepend_mask = torch.cat([prepend_mask, torch.ones((x.shape[0], 1), device=x.device, dtype=torch.bool)], dim=1)
+        if self.global_cond_type == "prepend":
+            if prepend_inputs is None:
+                # Prepend inputs are just the global embed, and the mask is all ones
+                prepend_inputs = global_embed.unsqueeze(1)
+                prepend_mask = torch.ones((x.shape[0], 1), device=x.device, dtype=torch.bool)
+            else:
+                # Prepend inputs are the prepend conditioning + the global embed
+                prepend_inputs = torch.cat([prepend_inputs, global_embed.unsqueeze(1)], dim=1)
+                prepend_mask = torch.cat([prepend_mask, torch.ones((x.shape[0], 1), device=x.device, dtype=torch.bool)], dim=1)
 
-        prepend_length = prepend_inputs.shape[1]
+            prepend_length = prepend_inputs.shape[1]
 
         x = self.preprocess_conv(x) + x
 
         x = rearrange(x, "b c t -> b t c")
 
-        if self.transformer_type == "local_attn":
-            if mask is not None:
-                mask = torch.cat([prepend_mask, mask], dim=1)
-            output = self.transformer(x, prepend_cond=prepend_inputs, cross_attn_cond=cross_attn_cond, cross_attn_cond_mask=cross_attn_cond_mask, mask=mask, **kwargs)
-        elif self.transformer_type == "x-transformers" or self.transformer_type == "continuous_transformer":
-            output = self.transformer(x, prepend_embeds=prepend_inputs, context=cross_attn_cond, context_mask=cross_attn_cond_mask, mask=mask, prepend_mask=prepend_mask, **kwargs)
+        extra_args = {}
+
+        if self.global_cond_type == "adaLN":
+            extra_args["global_cond"] = global_embed
+
+        if self.patch_size > 1:
+            x = rearrange(x, "b (t p) c -> b t (p c)", p=self.patch_size)
```

Comment:
Just checking - did you test that this patching transform does what's intended and perfectly cancels with the one on L.212? I just did a quick test and it seems like they might not:
```
import torch
from einops import rearrange

batch = 5
channels = 3
length = 24
patch_size = 4

x = torch.arange(0,length).unsqueeze(0).unsqueeze(-1)
x = x.expand(batch, -1, channels).clone()
x[:,:,0] = -1 * x[:,:,0] 
x[:,:,2] = 100

ref = x.clone()

x = rearrange(x, "b (t p) c -> b t (p c)", p=patch_size)
x = rearrange(x, "b t c -> b c t")
x = rearrange(x, "b (c p) t -> b c (t p)", p=patch_size)
x = rearrange(x, "b c t-> b t c") # needed to compare to original (we conv in the real code)

assert torch.all(ref == x)
```

Reply:
aha, the `p` and `c` should be swapped in the second half of this.